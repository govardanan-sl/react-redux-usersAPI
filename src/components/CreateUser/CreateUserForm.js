import { Grid, TextField } from '@material-ui/core';
import React from 'react'
import {useForm,Form} from '../useForm';

const initialValues ={
    name:'',
    job: ''
}


function CreateUserForm() {
    const { formData , handleInputChange} = useForm(initialValues);
    return (
        <Form>
            <Grid container>
                <Grid item xs={6}>
                    <TextField 
                        variant="outlined"
                        label="Name"
                        name="name"
                        value={formData&&formData.name}
                        onChange = {handleInputChange}
                    />
                    <TextField 
                        variant="outlined"
                        label="Job Role"
                        name="job"
                        value={formData&&formData.job}
                        onChange = {handleInputChange}
                    />
                </Grid>
                <Grid item xs={6}></Grid>
            </Grid>
        </Form>
    )
}

export default CreateUserForm;
