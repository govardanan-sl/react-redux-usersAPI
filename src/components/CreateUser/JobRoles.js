export const getJobRoles = () =>([
    {id:'1',title:'Web Developer'},
    {id:'2',title:'Fullstack Developer'},
    {id:'3',title:'Tester'}
])