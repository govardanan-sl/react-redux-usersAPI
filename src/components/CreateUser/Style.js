import { makeStyles } from "@material-ui/core";

const useStyle = makeStyles(theme=>({
    root:{
        '& .MuiFormControl-root':{
            width:'80%',
            margin:theme.spacing(1)
        }
    }
}))

export default useStyle;