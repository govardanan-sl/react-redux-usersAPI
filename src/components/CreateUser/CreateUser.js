import { makeStyles, Paper } from '@material-ui/core'
import { PeopleOutlineTwoTone } from '@material-ui/icons'
import React from 'react'
import PageHeader from '../Header/PageHeader'
import CreateUserForm from './CreateUserForm'

const useStyles = makeStyles(theme=>({
    pageContent:{
        margin:theme.spacing(5),
        padding:theme.spacing(3)
    }
}))

function CreateUser() {
    const classes = useStyles();
    return (
        <div>
            <PageHeader
                title="Create New User"
                subTitle="Create new users for a job" icon={<PeopleOutlineTwoTone fontSize="large"/>}/>
            <Paper className={classes.pageContent}>
                <CreateUserForm/>
            </Paper>
        </div>
    )
}

export default CreateUser
