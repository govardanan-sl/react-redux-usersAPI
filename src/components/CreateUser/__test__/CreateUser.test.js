import React from 'react';
import reactDom from 'react-dom';
import ReactDOM from 'react-dom';
import CreateUserForm from '../CreateUserForm';
import {cleanup, fireEvent, render} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import CreateUser from '../CreateUser';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("Renders",()=>{
    const div = document.createElement("div");
    ReactDOM.render(<CreateUserForm></CreateUserForm>,div);
    reactDom.unmountComponentAtNode(div);
});

it("updates on change",()=>{
    const {queryByPlaceholderText} = render(<CreateUser/>)
    const textField = queryByPlaceholderText("Name");
    fireEvent.change(textField,{target:{value:'Test'}})
    expect(textField.value).toBe("Test");
})


it("Reset Button Resets data",()=>{
    const {queryByPlaceholderText,queryByTestId} = render(<CreateUser/>)
    const resetButton = queryByTestId("resetButton");
    const textField = queryByPlaceholderText("Name");
    fireEvent.change(textField,{target:{value:'Test'}});
    expect(textField.value).toBe("Test");
    fireEvent.click(resetButton);
    expect(textField.value).toBe("");
})



it("Form Matches Snapshot",()=>{
    const tree = renderer.create(<CreateUser/>).toJSON();
    expect(tree).toMatchSnapshot();
})
