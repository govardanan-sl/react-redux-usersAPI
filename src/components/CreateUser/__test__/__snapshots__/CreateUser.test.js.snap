// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create User Form Matches Snapshot 1`] = `
<div>
  <div
    className="MuiPaper-root makeStyles-root-27 MuiPaper-elevation0"
  >
    <div
      className="makeStyles-pageHeader-28"
    >
      <div
        className="MuiPaper-root MuiCard-root makeStyles-pageIcon-29 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="9"
            cy="8.5"
            opacity=".3"
            r="1.5"
          />
          <path
            d="M4.34 17h9.32c-.84-.58-2.87-1.25-4.66-1.25s-3.82.67-4.66 1.25z"
            opacity=".3"
          />
          <path
            d="M9 12c1.93 0 3.5-1.57 3.5-3.5S10.93 5 9 5 5.5 6.57 5.5 8.5 7.07 12 9 12zm0-5c.83 0 1.5.67 1.5 1.5S9.83 10 9 10s-1.5-.67-1.5-1.5S8.17 7 9 7zm0 6.75c-2.34 0-7 1.17-7 3.5V19h14v-1.75c0-2.33-4.66-3.5-7-3.5zM4.34 17c.84-.58 2.87-1.25 4.66-1.25s3.82.67 4.66 1.25H4.34zm11.7-3.19c1.16.84 1.96 1.96 1.96 3.44V19h4v-1.75c0-2.02-3.5-3.17-5.96-3.44zM15 12c1.93 0 3.5-1.57 3.5-3.5S16.93 5 15 5c-.54 0-1.04.13-1.5.35.63.89 1 1.98 1 3.15s-.37 2.26-1 3.15c.46.22.96.35 1.5.35z"
          />
        </svg>
      </div>
      <div
        className="makeStyles-pageTitle-30"
      >
        <div
          className="MuiTypography-root MuiTypography-h6"
        >
          Create New User
        </div>
        <div
          className="MuiTypography-root MuiTypography-subtitle2"
        >
          Create new users for a job
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root makeStyles-pageContent-26 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <form
      autoComplete="off"
      className="makeStyles-root-31"
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          style={
            Object {
              "flexBasis": "100%",
              "maxWidth": "100%",
            }
          }
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
            data-testid="InputField"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
            >
              Name
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                name="name"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Name"
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-32 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-34"
                >
                  <span>
                    Name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
            >
              Job Role
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-job"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                id="mui-component-select-job"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                name="job"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value=""
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-32 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-34"
                >
                  <span>
                    Job Role
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "margin": "8px",
                }
              }
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Submit
              </span>
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
              data-testid="resetButton"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "margin": "8px",
                }
              }
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Reset
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
